lib/topics.js

```diff
@@ -8,18 +8,37 @@ const SNS    = require('./sns');
 class Topic {
 
   constructor(server, topicName) {
-    this._server = server;
-    this.name    = topicName;
+    this._server     = server;
+    this.name        = topicName;
+    this.subscribers = [];
   }
 
 
   // Publish to a topic.
   publish(message) {
-    return this.clientPromise
-      .then(client => {
-        const stringMessage = JSON.stringify(message);
-        return client.publish(this.name, stringMessage);
+    const stringMessage = JSON.stringify(message);
+
+    if (process.env.NODE_ENV === 'test') {
+      // JSON encode/decode the message before passing to
+      // subscriber, same transformation as when using SNS.
+      const parsedMessage = JSON.parse(stringMessage);
+      setImmediate(() => {
+        this.subscribers.forEach(handler => handler(parsedMessage));
       });
+      return Promise.resolve();
+    } else {
+      return this.clientPromise
+        .then(client => client.publish(this.name, stringMessage));
+    }
```

Comment:
> Application can subscribe/unsubscribe at will, as often as it needs to

I understand unsubscribe for correctness, but what's the real world use case?

> When the topic is configured dynamically, the the topic would use the broker API to add a subscription when the first subscriber is added

I think for now subscriptions should be done manually or by configuration.

The thing is, just like an email subscription, HTTP/S subscriptions must be confirmed. I think what Ironium needs to provide in the Connect middleware is useful logging so subscriptions can be easily confirmed (by seeing subscription requests in logs, together with the URL to confirm them).